server/rest/wms_source.py

```diff
@@ -20,20 +20,33 @@
 from girder.api import access
 from girder.api.describe import Description
 
+# A Quick implementation to call GetCapabilities
+from owslib.wms import WebMapService
+
 from girder.plugins.minerva.rest.source import Source
 
 
 class WmsSource(Source):
+
     def __init__(self):
         self.resourceName = 'minerva_source_wms'
         self.route('POST', (), self.createWmsSource)
 
     @access.user
     def createWmsSource(self, params):
+        print params
         name = params['name']
         baseURL = params['baseURL']
+        username = params['username'] if 'username' in params else None
+        password = params['password'] if 'password' in params else None
+        wms = WebMapService(baseURL, version='1.1.1',
+                            username=username,
+                            password=password
+                            )
+        layers = list(wms.contents)
         minerva_metadata = {
```

Comment:
We will want to call this this map service asynchronously in the future, but that will add complexity to the UI, so let's just track it with an issue for now.

https://github.com/Kitware/minerva/issues/112